lib/ruby_indexer/test/reference_finder_test.rb

```diff
@@ -0,0 +1,43 @@
+# typed: true
+# frozen_string_literal: true
+
+require "test_helper"
+
+module RubyIndexer
+  class ReferenceFinderTest < Minitest::Test
+    def test_finds_constant_references
```

Comment:
Can we have tests to cover more scenarios? Especially around singletons and top-level namespaces, like:

```rb
class Foo
  class << self
    Bar # should be a reference
  end
end
```

And maybe 

```rb
class Foo
  class ::Baz
    Bar # shouldn't be a reference
  end
end
```

Reply:
Added more tests 👍